### DISCUSSION TITLE
Thoughts on partially adopting the architecture

### DISCUSSION THREAD

**Discussion by DavidKmn** [pointfreeco/swift-composable-architecture#523] [General]
What are your thoughts on partially adopting the compose letter architecture for a subset of the app. We have a couple of large projects written in Viper/MVVM and would like to migrate/build new features using the composable architecture, any thoughts/examples/recommendations of how to approach this? @stephencelis @mbrandonw 

Thanks for your contributions to the iOS community 🙂

**Comment by andreyz** [comment]
I've migrated a production app piece by piece during several iterations from Coordinator Model View Controller style architecture. My approach was to rewrite a feature (views, logic) in TCA and then plug it with the rest of the app with a  ViewStore reference and sending actions to that reference where there were method calls previously. Once you've migrated the feature, write some units tests and really rip the benefits of TCA.
